File: reactjs-todo-dv/client/components/davinci-client/flow-button.js
/*
 * forgerock-sample-web-react
 *
 * flow-button.js
 *
 * Copyright (c) 2021 ForgeRock. All rights reserved.
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */
import React from 'react';

export default function FlowButton({ collector, flow, renderForm }) {

  const clickHandler = async () => {
    const node = await flow(collector.output.key);
    renderForm(node);
  };

  return (
    <button
      key={collector.output.key}
      type="button"
      className="btn btn-primary w-100 flow-link mb-2"
      onClick={clickHandler}
    >
      <span> {collector.output.label}</span>
    </button>
  );
}

File: android/kotlin-davinci/davinci/src/main/java/com/pingidentity/samples/app/davinci/collector/FlowButton.kt
/*
 * Copyright (c) 2024 - 2025 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.samples.app.davinci.collector

import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.wrapContentWidth
import androidx.compose.material3.Button
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.pingidentity.davinci.collector.FlowCollector

/**
 * The flow button.
 *
 * @param field The flow collector.
 * @param onNext The callback to be called when the next node is triggered.
 */
@Composable
fun FlowButton(
    field: FlowCollector,
    onNext: () -> Unit,
) {
    Row(
        modifier =
        Modifier
            .padding(4.dp)
            .fillMaxWidth(),
    ) {
        field.value = ""
        if (field.type == "FLOW_LINK") {
            TextButton(
                modifier =
                Modifier
                    .fillMaxWidth()
                    .wrapContentWidth(Alignment.CenterHorizontally),
                onClick = {
                    field.value = "action"
                    onNext()
                },
            ) {
                androidx.compose.material3.Text(field.label)
            }
        } else {
           Button(
                modifier =
                Modifier
                    .fillMaxWidth()
                    .wrapContentWidth(Alignment.CenterHorizontally),
                onClick = {
                    field.value = "action"
                    onNext()
                },
            ) {
                androidx.compose.material3.Text(field.label)
            }

        }
    }
}